src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -1,207 +1,375 @@
-            {
-                return;
-            }
-
-            FunctionResult result = await _triggerExecutor.ExecuteAsync(message, cancellationToken);
-            await _messageProcessor.CompleteProcessingMessageAsync(message, result, cancellationToken);
-        }
-
-        internal async Task ProcessSessionMessageAsync(IMessageSession session, Message message, CancellationToken cancellationToken)
-        {
-            _messageSession = session;
-            if (!await _sessionMessageProcessor.BeginProcessingMessageAsync(session, message, cancellationToken))
-            {
-                return;
-            }
-
-            FunctionResult result = await _triggerExecutor.ExecuteAsync(message, cancellationToken);
-            await _sessionMessageProcessor.CompleteProcessingMessageAsync(session, message, result, cancellationToken);
-        }
-
-        private void ThrowIfDisposed()
-        {
-            if (_disposed)
-            {
-                throw new ObjectDisposedException(null);
-            }
-        }
-
+        }
+
+        internal async Task ProcessMessageAsync(Message message, CancellationToken cancellationToken)
+        {
+            if (!await _messageProcessor.BeginProcessingMessageAsync(message, cancellationToken))
+            {
+                return;
+            }
+
+            ServiceBusTriggerInput input = ServiceBusTriggerInput.New(message);
+            input.MessageReceiver = Receiver;
+
+            FunctionResult result = await _triggerExecutor.TryExecuteAsync(GetTriggerFunctionData(message, input), cancellationToken);
+            await _messageProcessor.CompleteProcessingMessageAsync(message, result, cancellationToken);
+        }
+
+        internal async Task ProcessSessionMessageAsync(IMessageSession session, Message message, CancellationToken cancellationToken)
+        {
+            _messageSession = session;
+            if (!await _sessionMessageProcessor.BeginProcessingMessageAsync(session, message, cancellationToken))
+            {
+                return;
+            }
+
+            ServiceBusTriggerInput input = ServiceBusTriggerInput.New(message);
+            input.MessageSession = session;
+
+            FunctionResult result = await _triggerExecutor.TryExecuteAsync(GetTriggerFunctionData(message, input), cancellationToken);
+            await _sessionMessageProcessor.CompleteProcessingMessageAsync(session, message, result, cancellationToken);
+        }
+
+        internal void StartMultipleMessageListening(CancellationToken cancellationToken)
+        {
+            SessionClient sessionClient = null;
+            IMessageReceiver receiver = null;
+            if (_isSessionsEnabled)
+            {
+                sessionClient = _messagingProvider.CreateSessionClient(_entityPath, _serviceBusAccount.ConnectionString);
+            }
+            else
+            {
+                receiver = Receiver;

```

Comment:
For the receiver created here on either path, are all the AutoComplete/AutoRenew/etc. options in effect? How do those options affect batch processing?

Reply:
`AutoComplete/AutoRenew` are affected only single processing as the options can be passed to  `RegisterMessageHandler/RegisterSessionHandler`. `ReceiveAsync` does not use the options.

Reply:
Got it. So how will customers ensure their message locks are retained if they aren't being renewed? If they're in the middle of a long running function processing a large batch of messages, and the locks time out, another host may start processing these messages again concurrently.

Similarly for AutoComplete - what if the user wants/needs to manage complete/abandon/deadletter operations themselves for the messages.

Both of the above issues have come up a ton over the years in this extension.